mesop/examples/llm_playground.py

```diff
@@ -0,0 +1,504 @@
+import random
+import time
+from typing import Callable
+
+import mesop as me
+
+
+@me.stateclass
+class State:
+  title: str = "LLM Playground"
+  # Prompt / Response
+  input: str
+  response: str
+  # Tab open/close
+  prompt_tab: bool = True
+  response_tab: bool = True
+  # Model configs
+  selected_model: str = "gemini-1.5"
+  selected_region: str = "us-east4"
+  temperature: float = 0
+  token_limit: int = 1
+  stop_sequence: str = ""
+  stop_sequences: list[str]
+  # Modal
+  modal_open: bool = False
+  # Workaround for clearing inputs
+  clear_prompt_count: int = 0
+  clear_sequence_count: int = 0
+
+
+@me.page(path="/llm-playground", title="LLM Playground")
+def page():
+  state = me.state(State)
+
+  # Modal
+  with modal(modal_open=state.modal_open):
+    me.text("Get code", type="headline-5")
+    if "gemini" in state.selected_model:
+      me.text(
+        "Use the following code in your application to request a model response."
+      )
+      with me.box(style=_STYLE_CODE_BOX):
+        me.markdown(
+          _GEMINI_CODE_TEXT.format(
+            content=state.input.replace('"', '\\"'),
+            model=state.selected_model,
+            region=state.selected_region,
+            stop_sequences=make_stop_sequence_str(state.stop_sequences),
+            token_limit=state.token_limit,
+            temperature=state.temperature,
+          )
+        )
+    else:
+      me.text(
+        "You can use the following code to start integrating your current prompt and settings into your application."
+      )
+      with me.box(style=_STYLE_CODE_BOX):
+        me.markdown(
+          _GPT_CODE_TEXT.format(
+            content=state.input.replace('"', '\\"').replace("\n", "\\n"),
+            model=state.selected_model,
+            stop_sequences=make_stop_sequence_str(state.stop_sequences),
+            token_limit=state.token_limit,
+            temperature=state.temperature,
+          )
+        )
+    me.button(label="Close", type="raised", on_click=on_click_modal_bg)
+
+  # Main content
+  with me.box(style=_STYLE_CONTAINER):
+    # Main Header
+    with me.box(style=_STYLE_MAIN_HEADER):
+      with me.box(style=_STYLE_TITLE_BOX):
+        me.text(
+          state.title,
+          type="headline-6",
+          style=me.Style(line_height="24px", margin=me.Margin(bottom=0)),
+        )
+
+    # Toolbar Header
+    with me.box(style=_STYLE_CONFIG_HEADER):
+      icon_button(
+        icon="code", tooltip="Code", label="CODE", on_click=on_click_show_code
+      )
+
+    # Main Content
+    with me.box(style=_STYLE_MAIN_COLUMN):
+      # Prompt Tab
+      with tab_box(header="Prompt", key="prompt_tab"):
+        me.textarea(
+          label="Write your prompt here, insert media and then click Submit",
+          # Workaround: update key to clear input.
```

Comment:
As an aside (not blocking this PR), I think we should brainstorm a proper API for clearing inputs. It's quite tricky to figure this out and it's a relatively common use case.

One thing we could do is providing a command API (which you can call inside an event handler), e.g. 

```py
me.clear_input(key="prompt")
```

This would work well if the key is globally unique. If the key is not... then we'd probably fire a warning and clear either all, one or none of them.